### PR TITLE
Custom status bar

### LINKED ISSUE #45: ios time bar color

### PR DESCRIPTION
**Pull request by marynag**

### CODE REVIEW

File: src/components/PlantPhotoSwiper/PlantPhotoSwiper.tsx [hunk 1]
```diff
@@ -33,10 +40,15 @@ export const PlantPhotoSwiper = () => {
 		return <NotFound />;
 	}
 
+	if (data?.data.length) {
+		setHasPlanPhoto();
+	}
```
Comment (Taraslysak) [inline src/components/PlantPhotoSwiper/PlantPhotoSwiper.tsx]: state setting should be done in useEffects

Comment (Taraslysak) [reply src/components/PlantPhotoSwiper/PlantPhotoSwiper.tsx]: Why do we need to do this?

File: src/components/CustomStatusBar/CustomStatusBar.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { StatusBar } from 'react-native';
+
+type Props = {
+	textColor: 'light-content' | 'dark-content';
+};
+
+export const CustomStatusBar = ({ textColor }: Props) => {
+	return <StatusBar barStyle={textColor} />;
+};
```
Comment (Taraslysak) [inline src/components/CustomStatusBar/CustomStatusBar.tsx]: I can't see the point in using CustomStatusBar. It looks like using straight StatusBar is easier

File: src/components/CustomStatusBar/CustomStatusBar.constants.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+type StatusBarTextColorType = 'light-content' | 'dark-content';
+
+export const STATUS_BAR_TEXT_COLOR: { [key: string]: StatusBarTextColorType } = {
+	light: 'light-content',
+	dark: 'dark-content',
+};
```
Comment (Taraslysak) [inline src/components/CustomStatusBar/CustomStatusBar.constants.ts]: IMHO, redundant and over-complicated code

File: src/components/PlantDetailLayout/PlantDetailLayout.tsx [hunk 4]
```diff
@@ -1,26 +1,39 @@
...
+	const [hasPlanPhoto, setHasPlanPhoto] = useState<boolean>(false);
+
+	const setIsPhotoState = () => {
+		setHasPlanPhoto(true);
+	};
 
 	const plantUuid = uuid ?? '';
+
+	const statusBarHeight = useSafeAreaInsets().top;
+	const statusBarHeightToAdd = hasPlanPhoto ? 0 : statusBarHeight;
```
Comment (Taraslysak) [inline src/components/PlantDetailLayout/PlantDetailLayout.tsx]: Why do we need this?

**Review by Taraslysak — COMMENTED** [review]
Please, consider some refactoring

File: src/components/CustomStatusBar/CustomStatusBar.constants.ts (hunk 3 above)
Comment (Nazar-kob) [reply src/components/CustomStatusBar/CustomStatusBar.constants.ts]: !

File: src/components/CustomStatusBar/CustomStatusBar.tsx (hunk 2 above)
Comment (Nazar-kob) [reply src/components/CustomStatusBar/CustomStatusBar.tsx]: same thought

**Review by Nazar-kob — COMMENTED** [review]
need refactoring

File: src/components/PlantDetailLayout/PlantDetailLayout.tsx (hunk 4 above)
Comment (marynag) [reply src/components/PlantDetailLayout/PlantDetailLayout.tsx]: > Why do we need this?

We are considering the idea of not displaying the component at all if we don't have the necessary data. Therefore, the component located at the top should have additional padding for the status bar. We handle this by checking if the necessary information exists, and if it does, we add padding only to the component that appears at the top of the page

**Review by Nazar-kob — APPROVED** [review]

**Pull request merged**